tests/testcases/mod.rs

```diff
@@ -20,14 +21,16 @@ pub mod all {
     pub async fn key_value_works(controller: &dyn Controller) {
         async fn checks(
             metadata: AppMetadata,
-            test_init_key: String,
```

Comment:
Hi @itowlson, here I have tried to implement an alternative solution to verify random key/value pair. This is to keep the signature of `checks` function same across all the tests. but I am happy to revert my changes if you insist.

Reply:
This "random" generation is deterministic within a process, so could disguise a problem if the same code was run twice within the same process.  That's not an issue here, but could be if someone later extracted this to a library function, or if we started using "theory" style tests that were run with multiple inputs.

I am not sure what problem the change is aiming to solve.  Yes, the `checks` function itself ends up with a different signature.  But the closure being passed to `TestCaseBuilder::assertions` still has the right signature.  The `checks` function itself is an internal implementation detail of this test case, and should be free to accept whatever information it needs to perform the test rigorously.

To be clear, I've very happy to see improvements and simplifications to a rather messy piece of code.  And I'm happy that this works here and now.  But I'm wary of establishing this as a pattern and of "checks functions must all look the same" as a design constraint.  Happy to keep iterating on this though of course!

Reply:
Also, if we do choose to do it this way, I'd suggest doing it as a separate PR.  This PR now seems to contain more lines relating to "consistent `checks` signature" than to the stated "clean up state directory before test"...  *grin*

Reply:
it finally clicked !! 

you changed the arguments of checks function only. and modified the closure to reflect that. the `assertions` function can remain unchanged across other test cases.

Thank you @itowlson for explaining. I've now reverted my changes. this looks awesome.